app/workers/job-reaper.ts

```diff
@@ -60,43 +60,44 @@ export default class WorkflowReaper implements Worker {
     const oldRunningJobs: Job[] = [];
     let isDone = false;
     let page = 0;
-    while (!isDone) {
-      const result = await Job.notUpdatedForMinutes(db, env.reapableJobAgeMinutes, page);
-      if (result.data.length > 0) {
-        oldRunningJobs.push(...result.data);
-      } else {
-        isDone = true;
+    try {
+      while (!isDone) {
+        const result = await Job.notUpdatedForMinutes(db, env.reapableJobAgeMinutes, page);
+        if (result.data.length > 0) {
+          oldRunningJobs.push(...result.data);
+        } else {
+          isDone = true;
+        }
+        page += 1;
       }
-      page += 1;
-    }
 
-    // return any of the old jobs that have workflows that are terminated or deleted
-    const orphans = await getDeletedOrTerminatedJobs(oldRunningJobs, this.logger);
+      // return any of the old jobs that have workflows that are terminated or deleted
+      const orphans = await getDeletedOrTerminatedJobs(oldRunningJobs, this.logger);
 
-    // cancel the orphaned jobs
-    for (const job of orphans) {
-      this.logger.info(`Reaping job ${job.requestId}`);
-      job.cancel('Canceled by job reaper');
-      try {
-        await job.save(db);
-      } catch (e) {
-        this.logger.error(`Failed to cancel job ${job.requestId}`);
-        throw (e);
+      // cancel the orphaned jobs
+      for (const job of orphans) {
+        this.logger.info(`Reaping job ${job.requestId}`);
+        job.cancel('Canceled by job reaper');
+        try {
+          await job.save(db);
+        } catch (e) {
+          this.logger.error(`Failed to cancel job ${job.requestId}`);
+          throw (e);
+        }
       }
+    } catch (e) {
+      this.logger.error('Error while trying to cancel orphaned jobs');
+      this.logger.error(e);
     }
   }
 
   async start(): Promise<void> {
     this.isRunning = true;
     while (this.isRunning) {
+      await sleep(10000);
```

Comment:
See later comments in workflow-listener.ts.  It could be that there's a watchdog keeping this alive, in which case the second item probably doesn't need to be addressed, as long as we're assured the cleanup process keeps going if there's an error.

Reply:
I tested this with my postgres database not being set up for harmony (any time the postgres pod is destroyed I need to run the migrations to make things work), so I know the behavior changed from spinning non-stop to retrying every 10 seconds.

Reply:
I was wrong - it was dying after any exception - the periodic log messages were not from these workers. I'm changing it to catch the errors and continue on after sleeping the configured time period.